File: 26.Remove_Duplicates_From_Sorted_Array/main.py
class Solution:
    def removeDuplicates(self, nums):
        k = 0  # Index for the next unique element
        for i in range(1, len(nums)):
            if nums[i] != nums[k]:
                k += 1
                nums[k] = nums[i]
        return k + 1  # Return the count of unique elements

# Example usage
solution = Solution()
nums = [0, 0, 1, 1, 1, 2, 2, 3, 3, 4]
k = solution.removeDuplicates(nums)
print("Unique count:", k) # Output: 5
print("Modified array:", nums[:k]) # Output: [0, 1, 2, 3, 4]

File: 26.Remove_Duplicates_From_Sorted_Array/main.java
public class Solution {
    public int removeDuplicates(int[] nums) {
        int k = 0; // Index for the next unique element
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] != nums[k]) {
                k++;
                nums[k] = nums[i];
            }
        }
        return k + 1; // Return the count of unique elements
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        int[] nums = {0, 0, 1, 1, 1, 2, 2, 3, 3, 4};
        int k = solution.removeDuplicates(nums);
        System.out.println("Unique count: " + k); // Output: 5
        System.out.print("Modified array: ");
        for (int i = 0; i < k; i++) {
            System.out.print(nums[i] + " ");
        }
    }
}
